git_pile/cli.py

```diff
@@ -148,6 +151,16 @@ def parse_args(self, argv=sys.argv[1:]):
         return self.parser.parse_args(argv)
 
     def run(self, args):
+        if self.run_first_call:
```

Comment:
we could "disable"  the run() so we don't need the additional run_first_call?

```
>>> class B(object):
...     def foo(self):
...             self.foo = lambda *a, **k: None
...             print("x")
... 
>>> b = B()
>>> b.foo()
x
>>> b.foo()
>>> 
```

just add a comment on top like `# can be called only once`. In the case here this would be moved to a different function like `__update_config()`  since we want to disable only that part.

Reply:
oh... another way to do the same thing:

```
>>> class C(object):
...     @functools.cache
...     def foo(self):
...             print("foo")
... 
>>> c = C()
>>> c.foo()
foo
>>> c.foo()
>>>
```

Reply:
That wouldn't work because `PileCLI.run()` is in fact "recursive" (it is called by the `AmCmd`). The intention of `run_first_call` here is to do things that are supposed to happen only once. Maybe we could rename that variable to make the intention clearer? Or have a method like `run_one_time_setup()`?

Reply:
why doesn't it work if you move the part that is supposed to run only once to a method called __upate_config() that has this logic in it?

Reply:
I agree